docker-extension-test/src/test/java/org/ballerinax/docker/test/EnvVarTest.java

```diff
@@ -0,0 +1,111 @@
+
+import org.ballerinax.docker.exceptions.DockerPluginException;
+import org.ballerinax.docker.test.utils.DockerTestUtils;
+import org.ballerinax.docker.test.utils.ProcessOutput;
+import org.ballerinax.docker.utils.DockerPluginUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinax.docker.generator.DockerGenConstants.ARTIFACT_DIRECTORY;
+import static org.ballerinax.docker.test.utils.DockerTestUtils.getExposedPorts;
+
+/**
+ * Build with `buildImage` field set to false and check whether build command is shown.
+ */
+public class EnvVarTest {
+    private final Path sourceDirPath = Paths.get("src", "test", "resources", "docker-tests");
+    private final Path targetPath = sourceDirPath.resolve(ARTIFACT_DIRECTORY);
+    private final String dockerImage = "env:latest";
+
+    private void setEnv(Map<String, String> newEnv) throws Exception {
+        try {
+            Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
+            Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
+            theEnvironmentField.setAccessible(true);
+            Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
+            env.putAll(newEnv);
+            Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField(
+                    "theCaseInsensitiveEnvironment");
+            theCaseInsensitiveEnvironmentField.setAccessible(true);
+            Map<String, String> ciEnv = (Map<String, String>) theCaseInsensitiveEnvironmentField.get(null);
+            ciEnv.putAll(newEnv);
+        } catch (NoSuchFieldException e) {
+            Class[] classes = Collections.class.getDeclaredClasses();
+            Map<String, String> env = System.getenv();
+            for (Class cl : classes) {
+                if ("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
+                    Field field = cl.getDeclaredField("m");
+                    field.setAccessible(true);
+                    Object obj = field.get(env);
+                    Map<String, String> map = (Map<String, String>) obj;
+                    map.clear();
+                    map.putAll(newEnv);
+                }
+            }
+        }
+    }
+
+    @BeforeClass
+    public void compileSample() throws Exception {
+        Map<String, String> env = new HashMap<>();
+        env.put("PASSWORD", "myP@$$worD");
+        setEnv(env);
+        ProcessOutput buildProcess = DockerTestUtils.compileBallerinaFile(sourceDirPath, "env.bal");
```

Comment:
`DockerTestUtils.compileBallerinaFile()` function uses a process builder. I think we can pass env vars to the process so that the above `setEnv` function is not needed.